libraries/AP_Common/AP_Common.h

```diff
@@ -72,15 +73,43 @@
     inline uint8_t &operator[](size_t i) { return reinterpret_cast<uint8_t *>(this)[i]; }           \
     inline uint8_t operator[](size_t i) const { return reinterpret_cast<const uint8_t *>(this)[i]; }
 
-/*
-  check if bit bitnumber is set in value, returned as a
-  bool. Bitnumber starts at 0 for the first bit
+/**
+ * Check if bit bitnumber is set in value, returned as a
+ * bool. Bitnumber starts at 0 for the first bit
+ * @tparam T the variable type. It should be an integer type
+ * @param value the variable
+ * @param bitnumber bit number to be checked
+ * @return true if the bit is setted, false otherwise
  */
-#define BIT_IS_SET(value, bitnumber) (((value) & (1U<<(bitnumber))) != 0)
-
-// get high or low bytes from 2 byte integer
-#define LOWBYTE(i) ((uint8_t)(i))
-#define HIGHBYTE(i) ((uint8_t)(((uint16_t)(i))>>8))
+template <typename T>
```

Comment:
given compilation time cost of templates, I'd like to see a new AP_Common/bitops.h which we include only where needed

Reply:
done